server/text_generation_server/utils/layers.py

```diff
@@ -875,19 +878,16 @@ def load(cls, prefix, weights, eps=1e-6):
             return cls(weight, eps)
 
         def forward(self, hidden_states, residual=None):
-            if IS_XPU_SYSTEM:
-                residual_out = hidden_states
+            if IS_IPEX_AVAILABLE:
                 out = ipex.llm.functional.add_rms_norm(
                     residual,
                     hidden_states,
                     self.weight,
                     None,
                     self.variance_epsilon,
-                    True,
+                    residual is not None,
```

Comment:
What is this `True` flag ?

Is the call calling into FFI boundary (I'm wondering why the callsite cannot determine the potential None value of `residual` directly without the need for this extra flag)

Reply:
user could control if residual will be updated in place by this parameter.